### PR TITLE
New App: MARTA Map

### PR DESCRIPTION
**Pull request by InTheDaylight14**
Shows the real-time location of MARTA metro trains on a map. Offers two orientations, vertical or horizontal. 

In the horizontal mode this app has the option to show a station's estimated train arrival times. Not enough room for this in the vertical mode.

marta_map

### CODE REVIEW

File: apps/martamap/marta_map.star [hunk 1]
```diff
@@ -0,0 +1,806 @@
...
+                name = "Show Station arrivals?",
+                desc = "Show station arrival in the horrizontal orientation?",
+                icon = "toggle-on",
+                default = True,
+            ),
+            schema.Toggle(
+                id = "scroll",
+                name = "Scroll Arrival Head sign",
+                desc = "Scroll the arrival head signs or static text?",
+                icon = "toggle-on",
```
Comment (matslina) [inline apps/martamap/marta_map.star:98]:
```suggestion
                icon = "toggleOn",
```

File: apps/martamap/marta_map.star [hunk 2]
```diff
@@ -0,0 +1,806 @@
...
+                name = "Vertical Orientation?",
+                desc = "Show map in vertical orientation?",
+                icon = "rectangle-vertical",
+                default = False,
+            ),
+            schema.Toggle(
+                id = "arrivals",
+                name = "Show Station arrivals?",
+                desc = "Show station arrival in the horrizontal orientation?",
+                icon = "toggle-on",
```
Comment (matslina) [inline apps/martamap/marta_map.star:91]:
```suggestion
                icon = "toggleOn",
```

File: apps/martamap/marta_map.star [hunk 3]
```diff
@@ -0,0 +1,806 @@
...
+                name = "Scroll Arrival Head sign",
+                desc = "Scroll the arrival head signs or static text?",
+                icon = "toggle-on",
+                default = True,
+            ),
+            schema.Dropdown(
+                id = "station",
+                name = "Stations",
+                desc = "The color of text to be displayed.",
+                icon = "train-subway-tunnel",
```
Comment (matslina) [inline apps/martamap/marta_map.star:105]:
```suggestion
                icon = "trainSubway",
```

File: apps/martamap/marta_map.star [hunk 4]
```diff
@@ -0,0 +1,806 @@
...
+    direction_options = get_direction_list_options()
+
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Toggle(
+                id = "orientation",
+                name = "Vertical Orientation?",
+                desc = "Show map in vertical orientation?",
+                icon = "rectangle-vertical",
```
Comment (matslina) [inline apps/martamap/marta_map.star:84]: Not sure what to use here instead. Check out https://github.com/tidbyt/community/blob/main/apps/icons.go.

File: apps/martamap/marta_map.star [hunk 5]
```diff
@@ -0,0 +1,806 @@
...
+        print("MARTA Train API request failed with status %d", response.status_code)
+    all_trains = response.json()
+    all_train_arrivals = all_trains["RailArrivals"]
+
+    trains = {}
+    for arrival in all_train_arrivals:
+        if arrival["TRAIN_ID"] not in trains.keys():
+            trains[arrival["TRAIN_ID"]] = arrival
+
+    cache.set("cached_trains", json.encode(trains), ttl_seconds = 10)
```
Comment (matslina) [inline apps/martamap/marta_map.star:159]: Could we do a bit higher TTL?

**Review by matslina — CHANGES_REQUESTED** [review]
Thanks for the PR. Very good looking app! I live no where near a MARTA stop but still kinda want to run this on my Tidbyt. =)

There are a few issues remaining for the build to pass. Proposed a couple of icon fixes here.

File: apps/martamap/marta_map.star (hunk 5 above)
Comment (InTheDaylight14) [reply apps/martamap/marta_map.star]: Yeah, for sure

**Comment by InTheDaylight14** [comment]
> Thanks for the PR. Very good looking app! I live no where near a MARTA stop but still kinda want to run this on my Tidbyt. =)
> 
> There are a few issues remaining for the build to pass. Proposed a couple of icon fixes here.

Thank you for the review! I think I hit everything.

**Review by matslina — APPROVED** [review]

**Pull request merged**